--- src/web100-util.c	(revision 830)
+++ src/web100-util.c	(working copy)
@@ -647,10 +647,17 @@
                    int *AbruptTimeouts, int *SendStall, int *SlowStart,
                    int *SubsequentTimeouts, int *ThruBytesAcked) {
   int i;
+  int PktsIn = -1;
+  int DataPktsIn = -1;
+  int has_AckPktsIn = 0;
 
   for (i = 0; i <= count_vars; i++) {
     if (strcmp(web_vars[i].name, "Timeouts") == 0)
       *Timeouts = atoi(web_vars[i].value);
+    else if (strcmp(web_vars[i].name, "PktsIn") == 0)
+      PktsIn = atoi(web_vars[i].value);
+    else if (strcmp(web_vars[i].name, "DataPktsIn") == 0)
+      DataPktsIn = atoi(web_vars[i].value);
     else if (strcmp(web_vars[i].name, "SumRTT") == 0)
       *SumRTT = atoi(web_vars[i].value);
     else if (strcmp(web_vars[i].name, "CountRTT") == 0)
@@ -667,8 +674,10 @@
       *CurrentMSS = atoi(web_vars[i].value);
     else if (strcmp(web_vars[i].name, "DupAcksIn") == 0)
       *DupAcksIn = atoi(web_vars[i].value);
-    else if (strcmp(web_vars[i].name, "AckPktsIn") == 0)
+    else if (strcmp(web_vars[i].name, "AckPktsIn") == 0) {
+      has_AckPktsIn = 1;
       *AckPktsIn = atoi(web_vars[i].value);
+    }
     else if (strcmp(web_vars[i].name, "MaxRwinRcvd") == 0)
       *MaxRwinRcvd = atoi(web_vars[i].value);
     else if (strcmp(web_vars[i].name, "X_Sndbuf") == 0)
@@ -729,6 +738,11 @@
       *ThruBytesAcked = atoi(web_vars[i].value);
   }
 
+  if (!has_AckPktsIn) {
+    *AckPktsIn = PktsIn - DataPktsIn;
+    printf("Calculating AckPktsIn: %d - %d: %d\n", PktsIn, DataPktsIn, *AckPktsIn);
+  }
+
   return (0);
 }
 
